### ISSUE TITLE
Sending data issue

### ISSUE THREAD

**Issue by banbeiduyao** [nopnop2002/Arduino-STM32-CAN#63]
I am currently using one F405, one F103, and USB to CAN for testing. There is no problem transferring data between USB to CAN and F103. However, after connecting F405 to send data for a period of time, F103 cannot send data. The reason cannot be determined!
Screenshots of normal and abnormal situations
![e7c0651bdea610a1bf860f1d3fdaa39](https://github.com/nopnop2002/Arduino-STM32-CAN/assets/50175483/dc9e8ed5-7d2c-4896-85f5-aaf94f0f5721)
![910061234ba317e61b302793ac4d3f9](https://github.com/nopnop2002/Arduino-STM32-CAN/assets/50175483/a1d60c10-568e-488e-90db-149ce6238541)

**Comment by nopnop2002** [comment]
What exactly is USB to CAN?

**Comment by banbeiduyao** [comment]
CAN data can be monitored through a computer

**Comment by nopnop2002** [comment]
It works correctly when there are two nodes, USB to CAN and F103.

It works correctly when there are two nodes, USB to CAN and F405.

It works correctly when there are two nodes, F103 and F405.

However, it does not work correctly when there are three nodes: USB to CAN, F103, and F405.


Is this understanding correct?

**Comment by banbeiduyao** [comment]
Yes, as long as F405 is connected, F103 will fail to send data after a period of time, and the time may not be certain

**Comment by banbeiduyao** [comment]
Yes, as long as F405 is connected, F103 will fail to send data after a period of time, and the time may not be certain

**Comment by nopnop2002** [comment]
It works correctly when there are two nodes, USB to CAN and F103. ---> The CAN bus has a short stub length.

It works correctly when there are two nodes, USB to CAN and F405. ---> The CAN bus has a short stub length.

It works correctly when there are two nodes, F103 and F405. ---> The CAN bus has a short stub length.

However, it does not work correctly when there are three nodes: USB to CAN, F103, and F405. ---> The CAN bus stub length is a little longer.

---

The CAN bus stub length may be too long.
ISO11898 Standard specifies a maximum bus length of 40m and maximum stub length of 0.3m at 1Mbps. See [here](https://e2e.ti.com/support/interface-group/interface/f/interface-forum/378932/iso1050-can-bus-stub-length).


Or Lowering the Bit-Rate may solve the problem.

**Comment by banbeiduyao** [comment]
Thank you very much, but the wiring distance here is very short. F405>>5cm>>F103>>20cm>>USB-CAN

**Comment by nopnop2002** [comment]
The problem may be solved by lowering the Bit-Rate or changing the value of the terminating resistor.

The value of the terminating resistor is related to the length of the CAN-BUS.
However, obtaining accurate resistance values ​​requires extremely difficult calculations.

**Comment by banbeiduyao** [comment]
Okay, I'll test it again. Thank you

**Comment by nopnop2002** [comment]
If you have a mix of 3.3V and 5V transceivers, a +1V ground shift using split termination can be very effective.
Check out Figure 7 here.
https://www.ti.com/lit/an/slla337/slla337.pdf


---

Please check again if the Bus is as shown below.

Terminator--F405--5cm--F103--20cm--USB-CAN--Terminator
